lms/djangoapps/courseware/views.py

```diff
@@ -686,6 +687,26 @@ def course_info(request, course_id):
             'url_to_enroll': url_to_enroll,
         }
 
+        # Get the URL of the user's last position in order to display the 'where you were last' message
```

Comment:
Could this follow the same pattern as the contexts above?  Pull all this logic into its own method and just call

```Python
last_accessed_courseware = get_last_accessed(args)

context = {
   ...
   'last_accessed_courseware': last_accessed_courseware
}
```